script/vsts/platforms/templates/build.yml

```diff
@@ -0,0 +1,58 @@
+steps:
+  - pwsh: |
+      # OS specific env variables
+      if ($env:AGENT_OS -eq "Windows_NT") {
+        $env:SQUIRREL_TEMP="C:/tmp"
+        $env:npm_config_build_from_source=true
+        $ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL=$env:ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL
+        $ATOM_WIN_CODE_SIGNING_CERT_PASSWORD=$env:ATOM_WIN_CODE_SIGNING_CERT_PASSWORD
+      }
+      if ($env:AGENT_OS -eq "Darwin") {
+        $ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL=$env:ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL
+        $ATOM_MAC_CODE_SIGNING_CERT_PASSWORD=$env:ATOM_MAC_CODE_SIGNING_CERT_PASSWORD
+        $ATOM_MAC_CODE_SIGNING_KEYCHAIN=$env:ATOM_MAC_CODE_SIGNING_KEYCHAIN
+        $ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD=$env:ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD
+        $AC_USER=$env:AC_USER
+        $AC_PASSWORD=$env:AC_PASSWORD
+      }
+      if ($env:AGENT_OS -eq "Linux") {
+        $env:CC=clang
+        $env:CXX=clang++
+        $env:npm_config_clang=1
+        $env:LinuxArgs="--create-debian-package --create-rpm-package"
+        $env:SHOULD_SIGN="false"
+      }
+
+      # Build Arguments
+      ## Creation of Windows Installaer
+      if ($env:AGENT_OS -eq "Windows_NT") {
+        mkdir -f -p $env:SQUIRREL_TEMP
+        if ($env:IS_RELEASE_BRANCH -eq "true") {
+          $CreateWindowsInstallaer="--create-windows-installer"
+        }
+      }
+
+      ## Code Sign
+      if ( ($env:SHOULD_SIGN -eq "true") -and (($env:IS_RELEASE_BRANCH -eq "true")  -or ($env:IS_SIGNED_ZIP_BRANCH -eq "true")) ) {
+        $CodeSign="--code-sign"
+      }
+
+      # Build
+      $esc = '--%'
+      if (($env:AGENT_OS  -eq "Windows_NT") -and ($env:BUILD_ARCH -eq "x86")) {
+        node 'script\vsts\windows-run.js' 'script\build.cmd' --no-bootstrap --compress-artifacts $esc $CodeSign $CreateWindowsInstallaer
+      } else {
+        script/build --no-bootstrap --compress-artifacts $esc $env:LinuxArgs $CodeSign $CreateWindowsInstallaer
+      }
+    displayName: Build Atom
+    env:
+      GITHUB_TOKEN: $(GITHUB_TOKEN)
+      IS_RELEASE_BRANCH: $(IsReleaseBranch)
+      IS_SIGNED_ZIP_BRANCH: $(IsSignedZipBranch)
+      ATOM_RELEASE_VERSION: $(ReleaseVersion)
+
```

Comment:
In this `env:` block, we should additionally be passing in all of the following:
```yml
          ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL: $(ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL)
          ATOM_MAC_CODE_SIGNING_CERT_PASSWORD: $(ATOM_MAC_CODE_SIGNING_CERT_PASSWORD)
          ATOM_MAC_CODE_SIGNING_KEYCHAIN: $(ATOM_MAC_CODE_SIGNING_KEYCHAIN)
          ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD: $(ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD)
          ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL: $(ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL)
          ATOM_WIN_CODE_SIGNING_CERT_PASSWORD: $(ATOM_WIN_CODE_SIGNING_CERT_PASSWORD)